src/granite/connection_management.cr

```diff
@@ -51,6 +51,8 @@ module Granite::ConnectionManagement
     end
 
     def self.schedule_adapter_switch
+      return if @@writer_adapter == @@reader_adapter
```

Comment:
Not making a `spawn` call here saves ~8mb of memory allocation due to a fiber stack allocation, and saves me from my problem (and I'm hoping saves others from unnecessary memory allocation as well).